packages/contracts-bedrock/src/L2/SuperchainERC20.sol

```diff
@@ -1,49 +1,44 @@
+import { Unauthorized } from "src/libraries/errors/CommonErrors.sol";
 
 /// @title SuperchainERC20
 /// @notice SuperchainERC20 is a standard extension of the base ERC20 token contract that unifies ERC20 token
-///         bridging to make it fungible across the Superchain. It builds on top of the L2ToL2CrossDomainMessenger for
-///         both replay protection and domain binding.
-abstract contract SuperchainERC20 is ISuperchainERC20Extensions, ISuperchainERC20Errors, ERC20 {
-    /// @notice Address of the L2ToL2CrossDomainMessenger Predeploy.
-    address internal constant MESSENGER = Predeploys.L2_TO_L2_CROSS_DOMAIN_MESSENGER;
-
-    /// @notice Sends tokens to some target address on another chain.
-    /// @param _to      Address to send tokens to.
-    /// @param _amount  Amount of tokens to send.
-    /// @param _chainId Chain ID of the destination chain.
-    function sendERC20(address _to, uint256 _amount, uint256 _chainId) external virtual {
-        if (_to == address(0)) revert ZeroAddress();
-
-        _burn(msg.sender, _amount);
-
-        bytes memory _message = abi.encodeCall(this.relayERC20, (msg.sender, _to, _amount));
-        IL2ToL2CrossDomainMessenger(MESSENGER).sendMessage(_chainId, address(this), _message);
-
-        emit SendERC20(msg.sender, _to, _amount, _chainId);
+///         bridging to make it fungible across the Superchain. This construction allows the SuperchainTokenBridge to
+///         burn and mint tokens.
+abstract contract SuperchainERC20 is ERC20, ICrosschainERC20, ISemver {
+    /// @notice A modifier that only allows the SuperchainTokenBridge to call
+    modifier onlySuperchainTokenBridge() {
+        if (msg.sender != Predeploys.SUPERCHAIN_TOKEN_BRIDGE) revert Unauthorized();
+        _;
     }
 
-    /// @notice Relays tokens received from another chain.
-    /// @param _from   Address of the msg.sender of sendERC20 on the source chain.
-    /// @param _to     Address to relay tokens to.
-    /// @param _amount Amount of tokens to relay.
-    function relayERC20(address _from, address _to, uint256 _amount) external virtual {
-        if (msg.sender != MESSENGER) revert CallerNotL2ToL2CrossDomainMessenger();
+    /// @notice Semantic version.
+    /// @custom:semver 1.0.0-beta.1
+    function version() external view virtual returns (string memory) {
+        return "1.0.0-beta.1";
+    }
 
-        if (IL2ToL2CrossDomainMessenger(MESSENGER).crossDomainMessageSender() != address(this)) {
-            revert InvalidCrossDomainSender();
-        }
+    /// @notice Allows the SuperchainTokenBridge to mint tokens.
+    /// @param _to     Address to mint tokens to.
+    /// @param _amount Amount of tokens to mint.
+    function crosschainMint(address _to, uint256 _amount) external onlySuperchainTokenBridge {
+        _mint(_to, _amount);
 
-        uint256 source = IL2ToL2CrossDomainMessenger(MESSENGER).crossDomainMessageSource();
+        emit CrosschainMinted(_to, _amount);
```

Comment:
another naming nit, should this be present tense of `CrosschainMint` for consistency with the commong `methodName/EventNames` used in ERC20s of `transfer/Transfer` and `mint/Mint`? Guessing we've bikeshed on this before, so this (like my other comments so far) is not super important